### PR TITLE
RDM-6151- initial version of chart

### PR DESCRIPTION
**Pull request by akdasari**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: ci-values.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+useInterpodAntiAffinity: false
+
+global:
+  subscriptionId: "1c4f0704-a29e-403d-b719-b90c34ef14c9"
+  tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+  environment: aat
+dummyConfigInCI: true   #Need this for pipleine build only
+definitions:
+ - https://github.com/hmcts/ccd-data-store-api/raw/master/src/aat/resources/CCD_CNP_27_AUTOTEST1.xlsx
+definitionFilename: cmc-ccd.xlsx
```
Comment (smathangi) [inline ci-values.yaml]: This feature `definitionFilename` only works if the files are baked into the docker image at root level. I don't think latest definition-importer image has these files. May be we should remove this feature completely and offer only path based property "definitions" ?

Comment (akdasari) [reply ci-values.yaml]: ci-values is used to pass the pipeline, in actual scenario in CMC, they supply file at root level

File: ccd-definition-importer/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+All the common labels needed for the labels sections of the definitions.
+*/}}
+{{- define "ccddi.labels" }}
+app.kubernetes.io/name: {{ template "hmcts.ccddi.releaseName" . }}
+helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/instance: {{ template "hmcts.ccddi.releaseName" . }}
+{{- end -}}
+{{- define "importer.definition.vault" }}
+  {{- if eq .Values.global.subscriptionId "bf308a5c-0624-4334-8ff8-8dca9fd43783"}}
```
Comment (adusumillipraveen) [inline ccd-definition-importer/templates/_helpers.tpl]: This is hardcoding, shouldn't be the case.This should be a variable from values.

File: ccd-definition-importer/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  {{- "ccd-saat" -}}
+  {{- else }}
+  {{- "ccd-aat" -}}
+  {{- end }}
+{{- end }}
+
+{{- define "importer.definition.resourcegroup" }}
+  {{- if eq .Values.global.subscriptionId "bf308a5c-0624-4334-8ff8-8dca9fd43783"}}
```
Comment (adusumillipraveen) [inline ccd-definition-importer/templates/_helpers.tpl]: Hardcoding

File: ccd-definition-importer/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  {{- "ccd-shared-saat" -}}
+  {{- else }}
+  {{- "ccd-shared-aat" -}}
+  {{- end }}
+{{- end }}
+
+{{- define "importer.definition.vaultGit" }}
+  {{- if eq .Values.global.subscriptionId "bf308a5c-0624-4334-8ff8-8dca9fd43783"}}
+  {{- "infra-vault-sandbox" -}}
```
Comment (adusumillipraveen) [inline ccd-definition-importer/templates/_helpers.tpl]: Hardcoding

File: ccd-definition-importer/templates/definition-importer.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "hmcts.ccddi.releaseName" . }}
+  labels:
+    {{- ( include "ccddi.labels" . ) | indent 4 }}
+  annotations:
+    helm.sh/hook: post-install,post-upgrade
+    helm.sh/hook-weight: "-5"
```
Comment (adusumillipraveen) [inline ccd-definition-importer/templates/definition-importer.yaml]: Why is hook-weight changed? 
https://github.com/hmcts/chart-ccd/blob/3.x/ccd/templates/definition-importer.yaml#L13

**Review by adusumillipraveen — CHANGES_REQUESTED** [review]
Also, does this need a basic Readme? May be in future PRs

File: ccd-definition-importer/templates/_helpers.tpl (hunk 2 above)
Comment (akdasari) [reply ccd-definition-importer/templates/_helpers.tpl]: removed all hardcoded references

File: ccd-definition-importer/templates/_helpers.tpl (hunk 4 above)
Comment (akdasari) [reply ccd-definition-importer/templates/_helpers.tpl]: removed all hardcoded references

File: ccd-definition-importer/templates/definition-importer.yaml (hunk 5 above)
Comment (akdasari) [reply ccd-definition-importer/templates/definition-importer.yaml]: changed weight

File: ccd-definition-importer/templates/_helpers.tpl (hunk 3 above)
Comment (akdasari) [reply ccd-definition-importer/templates/_helpers.tpl]: removed all hardcoded references

File: ci-values.yaml (hunk 1 above)
Comment (smathangi) [reply ci-values.yaml]: Ok, found the config. It seems clients (CMC) have to build their own wrapper image on top of core definition-importer image and bake their definition files at root level. 
https://github.com/hmcts/cmc-ccd-domain/blob/master/definition/xlsx2json.Dockerfile#L5

**Review by smathangi — APPROVED** [review]

File: ci-values.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+useInterpodAntiAffinity: false
+
+global:
+  subscriptionId: "1c4f0704-a29e-403d-b719-b90c34ef14c9"
```
Comment (adusumillipraveen) [inline ci-values.yaml]: Not needed?

**Review by adusumillipraveen — APPROVED** [review]

File: ci-values.yaml (hunk 6 above)
Comment (akdasari) [reply ci-values.yaml]: removed

**Pull request merged**